examples/functorch/parallel_train_torchopt.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2022 MetaOPT Team. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+import argparse
+import math
+from collections import namedtuple
+from typing import Any, NamedTuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+import functorch
+import torchopt
+from functorch import combine_state_for_ensemble, grad_and_value, make_functional, vmap
+
+
+def make_spirals(n_samples, noise_std=0.0, rotations=1.0):
+    ts = torch.linspace(0, 1, n_samples, device=DEVICE)
```

Comment:
`DEVICE` is a global variable defined in `if __name__ == '__main__'`. Please pass this as function argument instead.

Reply:
Please read [zou example](https://github.com/pytorch/functorch/blob/main/examples/ensembling/parallel_train.py#L6).